### PR TITLE
Add test case to cover work submitted and released on non-existent node

### PR DESCRIPTION
**Pull request by amolgautam25**
Adding test cases for the issue : https://github.com/ansible/receptor/issues/317

Test Description: 
--- Start ```node1```
--- Submit work to ```node999``` ( non existent node )
--- ```node1``` shuts down starts again and releases the work

Expected result: 
The work is released

### CODE REVIEW

File: tests/functional/mesh/work_test.go [hunk 1]
```diff
@@ -536,6 +536,48 @@ func TestWork(t *testing.T) {
...
+			// start node 1
+			err = nodes["node1"].Start()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// release the work on node1
+			_, err = controllers["node1"].WorkRelease(unitID)
+			if err != nil {
+				t.Fatal(err)
```
Comment (amolgautam25) [inline tests/functional/mesh/work_test.go:568]: Getting an error here : 
 work_test.go:568: read unix @->/tmp/receptor-testing/controlsockets/254441336/controlsock: read: connection reset by peer
```
  work_test.go:568: write unix @->/tmp/receptor-testing/controlsockets/378275209/controlsock: write: broken pipe
--- FAIL: TestWork (0.00s)
```

File: tests/functional/mesh/work_test.go [hunk 2]
```diff
@@ -536,6 +536,58 @@ func TestWork(t *testing.T) {
...
+			ctx, _ = context.WithTimeout(context.Background(), 60*time.Second)
+			err = m.WaitForReady(ctx)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// release the work on node1
+			_, err = controllers["node1"].WorkRelease(unitID)
+			if err != nil {
+				t.Fatal(err)
```
Comment (amolgautam25) [inline tests/functional/mesh/work_test.go:585]: Getting an error here : 
```
     work_test.go:582: write unix @->/tmp/receptor-testing/controlsockets/506454955/controlsock: write: broken pipe
```

Comment (amolgautam25) [reply tests/functional/mesh/work_test.go:585]: It has been resolved now..

File: tests/functional/mesh/work_test.go [hunk 3]
```diff
@@ -536,6 +536,61 @@ func TestWork(t *testing.T) {
...
+			// submit work from node1 to non-existent-node
+			// node999 was never initialised
+			unitID, err := controllers["node1"].WorkSubmit("node999", "echosleeplong")
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// wait for 10 seconds
+			ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
+			err = m.WaitForReady(ctx)
```
Comment (fosterseth) [inline tests/functional/mesh/work_test.go]: Not sure we need this WaitForReady section -- the mesh is already up and running at this point. We do need the WaitForReady down below though, after the node restart.

Comment (fosterseth) [reply tests/functional/mesh/work_test.go]: that said, we probably want to make sure the work is in pending state

```golang
ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
err = controllers["node1"].AssertWorkPending(ctx, unitID)
if err != nil {
	t.Fatal(err)
}
```

Comment (amolgautam25) [reply tests/functional/mesh/work_test.go]: I have made the required changes

**Review by fosterseth — APPROVED** [review]

**Pull request merged**